src/tests/java/htsjdk/variant/variantcontext/filter/CompoundFilterTest.java

```diff
@@ -0,0 +1,76 @@
+package htsjdk.variant.variantcontext.filter;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.variantcontext.VariantContextBuilder;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Created by farjoun on 9/9/15.
+ */
+public class CompoundFilterTest {
+
+    static AllPassFilter pass = new AllPassFilter();
+    static AllFailFilter fail = new AllFailFilter();
+
+    static Allele refA = Allele.create("A", true);
+    static Allele G = Allele.create("G", false);
+
+    static VariantContext vc = new VariantContextBuilder("dummy", "chr1", 1, 1, Arrays.asList(refA, G)).make();
+
+    @DataProvider
+    Iterator<Object[]> testCompoundFilterProvider() {
+        final List<Object[]> filters = new ArrayList<Object[]>(10);
+
```

Comment:
Just to make the data provider a little more clear, I think it would be helpful to change the //all pass to a //requireAll=true or a //Passes all filters and then a //requireAll=false or similar below, only because it was initially confusing that there were two sections with the //one fail and //empty comments.